app/src/main/java/com/tungnk123/orpheus/media/FileSystemSongProvider.kt

```
@@ -0,0 +1,185 @@
+
+    private val uris = ConcurrentHashMap<String, Uri>()
+    private var explorer = SimpleFileSystem.Folder()
+    private val _isUpdating = MutableStateFlow(false)
+    val isUpdating = _isUpdating.asStateFlow()
+
+    private fun emitUpdate(value: Boolean) = _isUpdating.update { value }
+
+    @RequiresApi(Build.VERSION_CODES.S)
+    suspend fun fetch() {
+        emitUpdate(true)
+        try {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)